interfaces/IDeviceInfo.h

```diff
@@ -25,136 +25,315 @@
         enum { ID = ID_DEVICE_CAPABILITIES_AUDIO };
 
         virtual ~IDeviceAudioCapabilities() override = default;
 
         enum AudioOutput : uint8_t {
-            AUDIO_OTHER,
-            AUDIO_RF_MODULATOR,
-            AUDIO_ANALOG,
-            AUDIO_SPDIF, //over RCA or TOSLINK
-            AUDIO_HDMI0,
-            AUDIO_HDMI1,
-            AUDIO_DISPLAYPORT
+            AUDIO_OTHER = 0x00 /* @text: OTHER */,
+            AUDIO_RF_MODULATOR = 0x01 /* @text: RF_MODULATOR */,
+            AUDIO_ANALOG = 0x02 /* @text: ANALOG */,
+            AUDIO_SPDIF = 0x04 /* @text: SPDIF0 */, //over RCA or TOSLINK
+            AUDIO_HDMI0 = 0x08 /* @text: HDMI0 */,
+            AUDIO_HDMI1 = 0x10 /* @text: HDMI1 */,
+            AUDIO_DISPLAYPORT = 0x20 /* @text: DISPLAYPORT */
         };
 
         enum AudioCapability : uint8_t {
-            AUDIOCAPABILITY_NONE,
-            ATMOS,
-            DD,
-            DDPLUS,
-            DAD,
-            DAPV2,
-            MS12
+            AUDIOCAPABILITY_NONE = 0x00 /* @text: audiocapability_none */,
+            ATMOS = 0x01 /* @text: atmos */,
+            DD = 0x02 /* @text: dd */,
+            DDPLUS = 0x04 /* @text: ddplus */,
+            DAD = 0x08 /* @text: dad */,
+            DAPV2 = 0x10 /* @text: dapv2 */,
+            MS12 = 0x20 /* @text: ms12 */
         };
 
         enum MS12Capability : uint8_t {
-            MS12CAPABILITY_NONE,
-            DOLBYVOLUME,
-            INTELIGENTEQUALIZER,
-            DIALOGUEENHANCER
+            MS12CAPABILITY_NONE = 0x00 /* @text: ms12capability_none */,
+            DOLBYVOLUME = 0x01 /* @text: dolbyvolume */,
+            INTELIGENTEQUALIZER = 0x02 /* @text: inteligentequalizer */,
+            DIALOGUEENHANCER = 0x04 /* @text: dialogueenhancer */
         };
 
         enum MS12Profile : uint8_t {
-            MS12PROFILE_NONE,
-            MUSIC,
-            MOVIE,
-            VOICE
+            MS12PROFILE_NONE = 0x00 /* @text: ms12profile_none */,
+            MUSIC = 0x01 /* @text: music */,
+            MOVIE = 0x02 /* @text: movie */,
+            VOICE = 0x04 /* @text: voice */
         };
 
-        typedef RPC::IIteratorType<AudioOutput, ID_DEVICE_CAPABILITIES_AUDIO_OUTPUT> IAudioOutputIterator;
-        typedef RPC::IIteratorType<AudioCapability, ID_DEVICE_CAPABILITIES_AUDIO_CAPABILITY> IAudioCapabilityIterator;
-        typedef RPC::IIteratorType<MS12Capability, ID_DEVICE_CAPABILITIES_AUDIO_MS12_CAPABILITY> IMS12CapabilityIterator;
-        typedef RPC::IIteratorType<MS12Profile, ID_DEVICE_CAPABILITIES_AUDIO_MS12_PROFILE> IMS12ProfileIterator;
+        struct AudioOutputCaps {
+            IDeviceAudioCapabilities::AudioOutput audioOutput /* @brief Audio Output support */;
+            IDeviceAudioCapabilities::AudioCapability audioCapabilities /* @bitmask @brief Retrieves AudioCapabilities */;
+            IDeviceAudioCapabilities::MS12Capability ms12Capabilities /* @bitmask @brief Retrieves MS12 Capabilities */;
```

Comment:
I think ms12 support is not always available, so these could be OptionalType<>

Reply:
done